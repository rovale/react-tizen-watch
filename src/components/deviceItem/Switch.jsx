import React from 'react';
import PropTypes from 'prop-types';

const Switch = ({ device }) => (
  <div>
    <div className="ui-toggleswitch vertical-align-middle">
      <input
        type="checkbox"
        className="ui-switch-input"
        readOnly
        checked={device.attributes[0].value}
      />
      <div className="ui-switch-button" />
    </div>
    <div className="vertical-align-middle">
      &nbsp;{device.title}
    </div>
  </div>
);

Switch.propTypes = {
  device: PropTypes.shape({
    attributes: PropTypes.array.isRequired,
    title: PropTypes.string.isRequired,
  }).isRequired,
};

export default Switch;
