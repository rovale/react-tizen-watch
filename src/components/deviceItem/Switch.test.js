import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Switch from './Switch';

Enzyme.configure({ adapter: new Adapter() });

test('it should render as expected', () => {
  const device = {
    id: 'someDevice',
    title: 'Some Device',
    attributes: [{ value: true }],
  };
  const wrapper = mount((<Switch device={device} />)).children().first();
  expect(wrapper).toMatchSnapshot();
});
