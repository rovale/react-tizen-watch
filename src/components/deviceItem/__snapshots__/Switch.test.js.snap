// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render as expected 1`] = `
<div>
  <div
    className="ui-toggleswitch vertical-align-middle"
  >
    <input
      checked={true}
      className="ui-switch-input"
      readOnly={true}
      type="checkbox"
    />
    <div
      className="ui-switch-button"
    />
  </div>
  <div
    className="vertical-align-middle"
  >
    Â 
    Some Device
  </div>
</div>
`;
